File: Raven.Abstractions/Util/EscapingHelper.cs
// -----------------------------------------------------------------------
//  <copyright file="EscapingHelper.cs" company="Hibernating Rhinos LTD">
//      Copyright (c) Hibernating Rhinos LTD. All rights reserved.
//  </copyright>
// -----------------------------------------------------------------------
using System;
using System.Text;

namespace Raven.Abstractions.Util
{
	public static class EscapingHelper
	{
		 public static string EscapeLongDataString(string data)
		 {
			 const int limit = 65519; // limit in .net 4.5 for EscapeDataString

			 if (data.Length <= limit)
			 {
				 return Uri.EscapeDataString(data);
			 }

			 var result = new StringBuilder();

			 int loops = data.Length / limit;

			 for (int i = 0; i <= loops; i++)
			 {
				 if (i < loops)
				 {
					 result.Append(Uri.EscapeDataString(data.Substring(limit * i, limit)));
				 }
				 else
				 {
					 result.Append(Uri.EscapeDataString(data.Substring(limit * i)));
				 }
			 }

			 return result.ToString();
		 }

		 public static string UnescapeLongDataString(string data)
		 {
			 const int limit = 65519; // limit in .net 4.5 for EscapeDataString

			 if (data.Length <= limit)
			 {
				 return Uri.UnescapeDataString(data);
			 }

			 var result = new StringBuilder();

			 int loops = data.Length / limit;

			 for (int i = 0; i <= loops; i++)
			 {
				 if (i < loops)
				 {
					 result.Append(Uri.UnescapeDataString(data.Substring(limit * i, limit)));
				 }
				 else
				 {
					 result.Append(Uri.UnescapeDataString(data.Substring(limit * i)));
				 }
			 }

			 return result.ToString();
		 }
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/abstractions/util/EscapingHelper.java
package net.ravendb.abstractions.util;

import net.ravendb.client.utils.UrlUtils;


public class EscapingHelper {
  //hide constructor
  private EscapingHelper() {
  }

  public static String escapeLongDataString(String data) {
      int limit = 65519;

      if (data.length() <= limit) {
        return UrlUtils.escapeDataString(data);
      }

      StringBuilder result = new StringBuilder();

      int loops = data.length() / limit;

      for (int i = 0; i <= loops; i++) {
          if (i < loops) {
              result.append(UrlUtils.escapeDataString(data.substring(limit * i, limit)));
          } else {
              result.append(UrlUtils.escapeDataString(data.substring(limit * i)));
          }
      }

      return result.toString();
  }
}
